Makefile

```
@@ -125,23 +128,27 @@ start-integration: web/src/build/static/app.js bin/goalert bin/psql-lite bin/wai
 jest: node_modules 
 	yarn workspace goalert-web run jest $(JEST_ARGS)
 
-test: node_modules jest
+test: node_modules jest ## Run all unit tests
 	go test -short ./...
 
 force-yarn:
 	yarn install --no-progress --silent --frozen-lockfile --check-files
 
-check: force-yarn generate node_modules
-	# go run ./devtools/ordermigrations -check
-	go vet ./...
-	go run github.com/gordonklaus/ineffassign ./...
-	CGO_ENABLED=0 go run honnef.co/go/tools/cmd/staticcheck ./...
+check: check-go check-js ## Run all lint checks
+	./devtools/ci/tasks/scripts/codecheck.sh
+
+check-js: force-yarn node_modules
```

With `check-js` and `check-go` as separate tasks, `generate` must be a dependency on both since it generates both Go and js/ts. `make` will handle only running the task once but both tasks need to list it as a dependency.

Technically it should probably be a dependency of `check` _also_ but as long as it's on the other two, then `check`, `check-go`, and `check-js` will all behave correctly.

```suggestion
check-js: force-yarn node_modules generate
```

---

Wouldn't that execute generate multiple times on `make check`?  what about adding a check if generate has already been run to prevent multiple runs?

---

Actually, in my testing it didn't really add that much time.  Added `generate` to `check-js`

---

`make` will only run the dependant task once per invocation, so it will just serve to ensure it always gets run, and in the correct order. Likewise, if you run targets in parallel with `-j` it would wait for generate complete before running the go, and js checks